### DISCUSSION TITLE
Bank-Vaults migration complete!

### DISCUSSION THREAD

**Discussion by sagikazarmark** [bank-vaults/bank-vaults#2118] [Announcements]
Couple months ago we announced some changes in the project. Most notably the fact that we were migrating the project from the `banzaicloud` organization on GitHub to its own organization called `bank-vaults`.

We are happy to announce that the migration is now complete.

Although we attempted to minimize breaking changes resulting from the migration, there are certain changes you need to account for when upgrading from previous versions. We prioritized backward compatibility in most cases, so the changes should be minimal.

Also, please keep in mind that we are still actively updating the documentation. If you find anything inaccurate, feel free to submit an issue or a PR to the [documentation repository](https://github.com/bank-vaults/bank-vaults.dev).

Should you have any questions, feel free to reach out to us at one of the following channels:

- Comment here
- [GitHub Discussions](https://github.com/orgs/bank-vaults/discussions)
- [Slack](https://outshift.com/slack) (#bank-vaults channel)
- Email: [team@bank-vaults.dev](mailto:team@bank-vaults.dev)

### Components split up into separate repositories

Bank-Vaults was always an umbrella project for a number of different tools working with Hashicorp Vault.

During the migration, we decided to move these components into separate repositories. We expect that it's going to make contributing to these projects easier. Another outcome we hope is that we can tag new releases more quickly.

### Documentation moved

The documentation has been moved from the [Banzai Cloud website](https://banzaicloud.com/) to its own location at [https://bank-vaults.dev](https://bank-vaults.dev).

### Artifact location changes

Naturally, artifact locations changed with the migration as well.
This is mostly just a change in location, but in some cases, installation methods may have changed as well.

#### Container images

Container images are now exclusively available on `ghcr.io`, i.e., images are no longer pushed to Docker Hub.

You can find all of the images under `ghcr.io/bank-vaults`. Check out the [Packages](https://github.com/orgs/bank-vaults/packages) page on GitHub.

#### Helm charts

Previously, Helm charts were published in Banzai Cloud's [Helm chart repository](https://github.com/banzaicloud/banzai-charts/).

To simplify maintenance, we have decided to release Helm charts as OCI artifacts going forward.

You can find the Helm charts under `ghcr.io/bank-vaults/helm-charts`. Check out the [Packages](https://github.com/orgs/bank-vaults/packages) page on GitHub.

Installing Helm charts from OCI operates slightly differently from Helm chart repositories. Firstly, there's no need to add repositories anymore.

Secondly, Helm charts can be referenced using their OCI address:

```shell
helm upgrade --install --wait --namespace vault-system --create-namespace vault-operator oci://ghcr.io/bank-vaults/helm-charts/vault-operator
```

#### Binaries [BREAKING CHANGE]

Previously, binaries were available on the Bank-Vaults' repository Releases page. Although _this_ hasn't changed, the repository's location has. Furthermore, to keep prior references and imports intact, we decided to leave a fork of the Bank-Vaults repository at its original location.

As a result, links pointing to the Releases page of the old repository no longer work. To download old releases, you need to modify the download link:

```diff
# Release page
- https://github.com/banzaicloud/bank-vaults/releases/tag/1.19.0
+ https://github.com/bank-vaults/bank-vaults/releases/tag/1.19.0

# Artifact download link
- https://github.com/banzaicloud/bank-vaults/releases/download/1.19.0/bank-vaults-darwin-amd64.tar.gz
+ https://github.com/bank-vaults/bank-vaults/releases/download/1.19.0/bank-vaults-darwin-amd64.tar.gz
```

Additionally, `vault-env` binaries are no longer published under the Bank-Vaults repository. Going forward, binaries for new releases will be published on the [Releases](https://github.com/bank-vaults/vault-env/releases) page of the `vault-env` repository.

### Package import path changes

As a result of the migration, there are some trivial changes in import paths:

- All import paths start with `github.com/bank-vaults`
- Components moved out of the Bank-Vaults repository now have their own import paths

Although most of these projects are rarely imported, one exception is the vault-sdk (that also has been moved out of the Bank-Vaults repository):

```diff
- import "github.com/banzaicloud/bank-vaults/pkg/sdk"
+ import "github.com/bank-vaults/vault-sdk"
```

### New tagging scheme

As of Bank-Vaults CLI 1.30.0 (not released yet), Git tags will receive a `v` prefix. You can read more about this change [here](https://github.com/orgs/bank-vaults/discussions/2065).

### Old issue and PR references

Old issue and PR references are most probably broken due to leaving a copy at the original location which means those references now lead to a 404 page on GitHub.

References can easily be fixed by changing the URL:

```diff
- import "github.com/banzaicloud/bank-vaults/issues/2040"
+ import "github.com/bank-vaults/bank-vaults/issues/2040"
```

**Comment by eskwor** [comment]
We are trying to integrate vault-secrets-webhook with OpenBao. Before investing too much time in testing, we’d like to confirm whether this integration is feasible. Any input would be greatly appreciated.
